client/lib/interval/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { PropTypes } from 'react';
+
+/**
+ * Calls a given function on a given interval
+ */
+export default React.createClass( {
+	displayName: 'Interval',
+
+	propTypes: {
+		onTick: PropTypes.func.isRequired,
+		milliseconds: PropTypes.number,
+		leading: PropTypes.bool,
+		pauseWhenHidden: PropTypes.bool,
+		children: PropTypes.element
+	},
+
+	getDefaultProps: () => ( {
+		milliseconds: 30000,
+		leading: true,
+		pauseWhenHidden: true
+	} ),
+
+	getInitialState: () => ( {
+		timer: null
+	} ),
+
+	componentDidMount() {
+		this.start()
+	},
+
+	componentWillUnmount() {
+		this.stop()
+	},
+
+	componentDidUpdate() {
+		this.start();
+	},
+
+	run() {
+		clearTimeout( this.state.timer );
+
+		if ( document.hidden && this.props.pauseWhenHidden ) {
+			return this.setState( { timer: null } );
+		}
+
+		this.setState( { timer: setTimeout( this.run, this.props.milliseconds ) } );
+
+		this.props.onTick();
+	},
+
+	start() {
+		return ! this.state.timer && this.setState( {
```

Comment:
`setState` doesn't return a value so if the objective here is to indicate that the timer has started and the state has been set then maybe you should add a callback as a parameter for the `start` method and use it when setting the state.